### ISSUE TITLE
Full width Radio labels

### ISSUE THREAD

**Issue by TomPretty** [guardian/source#662]
**Is your feature request related to a problem? Please describe.**

The contributions landing page has this payment method selector

<img width="603" alt="Screenshot 2020-12-17 at 16 15 15" src="https://user-images.githubusercontent.com/17720442/102513556-43f5ae80-4083-11eb-956a-a1450c0e1469.png">

As it stands, I'm not sure if there's a great way to implement this with `src-radio`? Crucially we want the whole area between the label and the payment icon clickable. Our current solution requires internal knowledge the `Radio` component and broke when a `div` swapped for a `span` (#538)! We'd like a someway of building this in a less hacky way! 

**Describe the solution you'd like**
One potential solution could be something like a `fullWidthLabel` prop on the `Radio` component that causes it to fill it's parent width?

**Comment by SiAdcock** [comment]
Good point @TomPretty! Thanks for raising 🙂  I shall raise this with @akemitakagi and see if we can think what the best solution might be.

**Comment by SiAdcock** [comment]
I'd be interested to hear how you've implemented this. Are you absolutely positioning the icons or something?

Maybe instead of the radio label accepting a string, it could accept a `ReactNode`. Then you can pass whatever you like as the label.

Possible examples:

```tsx
<RadioGroup name="payment-method" label="Payment Method">
  <Radio value="direct-debit">
    <div css={flex}>
      <span>Direct debit</span>
      <SvgDirectDebit/>
  </Radio>
</RadioGroup>
```


```tsx
<RadioGroup name="payment-method" label="Payment Method">
  <Radio 
    value="direct-debit" 
    label={(
      <div css={flex}>
        <span>Direct debit</span>
        <SvgDirectDebit/>
    )} 
  />
</RadioGroup>
```

**Comment by TomPretty** [comment]
I think the label does accept a `ReactNode` at the moment, the issue that we faced is that it's rendered in this:

```jsx
 const LabelText = ({
	hasSupportingText,
	children,
}: {
	hasSupportingText?: boolean
	children: ReactNode
}) => {
	return (
		<span
			css={(theme) => [
				hasSupportingText ? labelTextWithSupportingText : "",
				labelText(theme.radio && theme),
			]}
			className="src-radio-label-text"
		>
			{children}
		</span>
	)
}
```

And we need that span to be full width. Our current hacky solution is to pass in some css to the `cssOverrides` prop of the `Radio` component that specifically targets that span.

**Comment by SiAdcock** [comment]
I see, thanks for clarifying that!

I can see no reason why the `LabelText` needs to be a `<span>`, that looks like unnecessary tinkering on my part. The surrounding `<label>` is block level anyway, so it already fills the container. 

I can restore `LabelText` to be a `<div>` again, and explicitly set the width to be 100%. If that would work for you, I can try and get a fix in quick.

**Comment by TomPretty** [comment]
Yeah that sounds like it would work perfectly for us 👍

**Comment by SiAdcock** [comment]
This is fixed in [2.8.1](https://github.com/guardian/source/releases/tag/2.8.1). Please let me know if you notice any issues
